ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -518,48 +510,41 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
             tap((errorMessage) => this.stateUpdates$.next({formErrorMessages: [errorMessage]}))
         );
 
-        const removeTempAddressOperator = () => (val$: Observable<any>) => val$.pipe(
-            withLatestFrom(this.state$),
-            filter(([_, state]) => !!state.activityInstanceGuid),
-            tap(() => busyCounter$.next(1)),
-            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
-            catchError(() => {
-                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
-                return of(null);
-            }),
-            tap(() => busyCounter$.next(-1))
-        );
-
         const canSaveRealAddress = (address: Address | null) =>
             this.enoughDataToSave(address) && this.meetsActivityRequirements(address);
 
-        // "Real" as opposed to "Temp"
+        // "Real" as opposed to "Temp". Important we return the saved address as properties are added on server
         const saveRealAddressAction$ = this.saveTrigger$.pipe(
             withLatestFrom(this.formErrorMessages$, currentAddress$),
             filter(([_, formErrors, addressToSave]) => {
-                const canSave = canSaveRealAddress(addressToSave);
-                if (!canSave) {
-                    this.block.hasValidAddress = false;
-                }
-                return !formErrors?.length && canSave;
+                return !formErrors?.length && canSaveRealAddress(addressToSave);
             }),
             tap(() => busyCounter$.next(1)),
             concatMap(([_, formErrors, addressToSave]) => this.addressService.saveAddress(addressToSave, false)),
-            tap(() => busyCounter$.next(-1)),
             removeTempAddressOperator(),
```

Comment:
Seems like it should be removed as well (the operator is removed at line 521)

Reply:
b80a3d821e7a653381ed4603d6b37d767d247c33